website/i18n/zh/docusaurus-plugin-content-blog/2022/07/04/apisix-integrates-with-hydra.md

```diff
@@ -0,0 +1,286 @@
+  oryd/hydra:v1.10.6 \
+  clients create --skip-tls-verify \
+    --id facebook-photo-backup \
+    --secret some-secret \
+    --grant-types authorization_code,refresh_token,client_credentials,implicit \
+    --response-types token,code,id_token \
+    --scope openid,offline,photos.read \
+    --callbacks http://127.0.0.1:9010/callback
+```
+
+以下示例将执行一个 OAuth 2.0 授权流程。为简化此操作，Hydra CLI 提供了一个名为 `hydra token user` 的辅助命令。
+
+```shell
+ docker run --rm -it \
+  --network hydraguide \
+  -p 9010:9010 \
+  oryd/hydra:v1.10.6 \
+  token user --skip-tls-verify \
+    --port 9010 \
+    --auth-url https://localhost:5444/oauth2/auth \
+    --token-url https://localhost:5444/oauth2/token \
+    --client-id facebook-photo-backup \
+    --client-secret some-secret \
+    --scope openid,offline,photos.read
+```
+
+返回结果如下则表示配置正常：
+
+```shell
+Setting up home route on http://127.0.0.1:9010/
+Setting up callback listener on http://127.0.0.1:9010/callback
+Press ctrl + c on Linux / Windows or cmd + c on OSX to end the process.
+If your browser doesn't open automatically, navigate to:
+
+        http://127.0.0.1:9010/
+```
+
+该步骤将会启动一个用户登录程序，我们将在下面的步骤中使用该程序。
+
+### 步骤五：启动 APISIX 并配置路由
+
+如果你还没有安装 APISIX，请参考：[安装 APISIX](https://apisix.apache.org/zh/docs/apisix/next/installation-guide)。
+
+安装完成后，仅需创建路由并配置 `openid-connect` 插件即可。为了更直观地验证效果，我们还需要启动了一个 NGINX 服务当做上游，你也可以使用已有的上游服务。
+
+:::note 注意
+
+本示例中的 APISIX 是安装在宿主机上，如果是使用 Docker 启动的 APISIX，可能会出现网络问题。
+
+:::
+
+1. 首先使用 Docker 创建一个 NGINX 上游。
+
+```shell
+docker run -d --name test-nginx -p 8083:80 nginx
+```
+
+2. 通过以下命令创建路由，并配置 `openid-connect` 插件。
+
+其中 `client_id` 和 `client_secret` 和 `scope` 是步骤四中设置的 ID。你可以参考 [Preparing for Production](https://www.ory.sh/docs/hydra/production#exposing-administrative-and-public-api-endpoints) 获取其他信息。从上述链接的内容中，我们可以看到 `discovery` 的地址为 `https://{IP:Port}/.well-known/openid-configuration`。
+
+```shell
+curl http://127.0.0.1:9080/apisix/admin/routes/1 \
+-H 'X-API-KEY: edd1c9f034335f136f87ad84b625c8f1' -X PUT -d '
+{
+    "uri":"/*",
+    "plugins":{
+        "openid-connect":{
+            "client_id":"facebook-photo-backup",
+            "client_secret":"some-secret",
+            "discovery":"https://127.0.0.1:5444/.well-known/openid-configuration",
+            "scope":"openid",
+            "token_endpoint_auth_method": "client_secret_basic",
+            "bearer_only": false,
+            "redirect_uri":"http://127.0.0.1:9080/callback"
+        }
+    },
+    "upstream":{
+        "type":"roundrobin",
+        "nodes":{
+            "127.0.0.1:8083":1
+        }
+    }
+}'
+```
+
+### 步骤六：访问 APISIX
+
+1. 在浏览器中输入 `http://127.0.0.1:9080/index.html`，由于已经开启了 `openid-connect` 插件，所以页面被重定向到登录页面。在此页面输入用户验证程序中默认的账号密码。
+
+![网络错误/认证页面](https://static.apiseven.com/2022/blog/0704/1.png)
+
+2. 选择认证协议，并单击 `Allow Access`。
+
+![网络错误/选择页面](https://static.apiseven.com/2022/blog/0704/2.png)
+
+3. 验证成功之后，就可以访问到上游服务的页面。
+
+![网络错误/上游页面](https://static.apiseven.com/2022/blog/0704/3.png)
+
```

Comment:
Duplicate step 2, need to fix it. Use tab to indent pictures into each step should work perfectly for you.
<img width="1920" alt="image" src="https://user-images.githubusercontent.com/36651058/177235683-2d9f928e-c6e2-4696-bce2-6b1b4b4ba443.png">